app/controllers/concerns/with_organization.rb

```diff
@@ -1,11 +1,18 @@
 module WithOrganization
   def set_organization!
-    Organization.find_by!(name: organization_name).switch!
+    organization = Organization.by_custom_domain(request.domain) || Organization.find_by!(name: organization_name)
+    organization.switch!
   rescue => e
     Organization.central.switch!
     raise e
   end
 
+  def set_cookie_domain!
+    Mumukit::Login.configure do |config|
+      config.mucookie_domain = Organization.current.settings.laboratory_custom_domain || ENV['MUMUKI_COOKIES_DOMAIN'] || ENV['MUMUKI_MUCOOKIE_DOMAIN']
```

Comment:
@flbulgarelli we need to find a better way of updating cookies domain in each request... this is ugly.